ql/src/java/org/apache/hadoop/hive/ql/optimizer/SharedWorkOptimizer.java

```diff
@@ -326,6 +372,7 @@ public boolean sharedWorkOptimization(ParseContext pctx, SharedWorkOptimizerCach
                 LOG.debug("{} and {} cannot be merged", retainableTsOp, discardableTsOp);
                 continue;
               }
+              // FIXME: I think this optimization is assymetric; but the check is symmetric
```

Comment:
this could be done as a cleanup - however I've already concluded that because of the table ordering the problematic case will actually never happen; so we are safe

I've removed the FIXME